Support collection signing verification (#357)
ansible/ansible-builder: docs/usage.rst

   $ ansible-builder build --file=my-ee.yml

> ``--galaxy-keyring``

Reviewer: Need some docs for the other new options.

   $ ansible-builder build --file=my-ee.yml

``--galaxy-keyring``
********************

With more recent versions of Ansible, it is possible to have the ``ansible-galaxy``
utility verify collection signatures during installation. This requires a keyring to
be provided (can be built with GnuPG tooling) to use during verification. Provide
the path to this keyring with the ``--galaxy-keyring`` option. If this option is not
supplied, no signature verification will be performed. If it is provided, and the
version of Ansible is not recent enough to support this feature, an error will
occur during the image build process.

.. code::

   $ ansible-builder create --galaxy-keyring=/path/to/pubring.kbx
   $ ansible-builder build --galaxy-keyring=/path/to/pubring.kbx

``--galaxy-ignore-signature-status-code``
*****************************************

> With ``--galaxy-keyring ``set it is possible to ignore certain errors that may occur while while verifying collections.

Reviewer: ```suggestion
With ``--galaxy-keyring`` set it is possible to ignore certain errors that may occur while while verifying collections.
```

> With ``--galaxy-keyring`` set it is possible to ignore certain errors that may occur while while verifying collections.

Reviewer 2: s/while while/while/ ;)

It is passed unmodified to ``ansible-galaxy`` calls via the option ``--ignore-signature-status-code``.
See the ``ansible-galaxy`` documentation for more information.

.. code::

   $ ansible-builder create --galaxy-keyring=/path/to/pubring.kbx --galaxy-ignore-signature-status-code 500
   $ ansible-builder build --galaxy-keyring=/path/to/pubring.kbx --galaxy-ignore-signature-status-code 500

``--galaxy-required-valid-signature-count``
*******************************************

> With ``--galaxy-keyring`` set it is possible specify constrainsts of collection validation. The value is

Reviewer 2: ```suggestion
When ``--galaxy-keyring`` is set, the number of required valid collection signatures can be overridden. The value is
```

   $ ansible-builder build --file=my-ee.yml

``--galaxy-keyring``
********************

With more recent versions of Ansible, it is possible to have the ``ansible-galaxy``
utility verify collection signatures during installation. This requires a keyring to
be provided (can be built with GnuPG tooling) to use during verification. Provide
the path to this keyring with the ``--galaxy-keyring`` option. If this option is not
supplied, no signature verification will be performed. If it is provided, and the
version of Ansible is not recent enough to support this feature, an error will
occur during the image build process.

.. code::

   $ ansible-builder create --galaxy-keyring=/path/to/pubring.kbx
   $ ansible-builder build --galaxy-keyring=/path/to/pubring.kbx

``--galaxy-ignore-signature-status-code``
*****************************************

> With --galaxy-keyring set it is possible to ignore certain errors happening while

Reviewer: ```suggestion
With ``--galaxy-keyring`` set, it is possible to ignore certain errors that may occur while
```

> verifying collections. It is passed unmodified to ansible-galaxy calls via the option

Reviewer: ```suggestion
verifying collections. It is passed unmodified to ``ansible-galaxy`` calls via the option
```

> --ignore-signature-status-code. See the ansible-galaxy doc for more info.

Reviewer: ```suggestion
``--ignore-signature-status-code``. See the ``ansible-galaxy`` documentation for more information.
```

.. code::

   $ ansible-builder create --galaxy-keyring=/path/to/pubring.kbx --galaxy-ignore-signature-status-code 500
   $ ansible-builder build --galaxy-keyring=/path/to/pubring.kbx --galaxy-ignore-signature-status-code 500

``--galaxy-required-valid-signature-count``
*******************************************

> With --galaxy-keyring set it is possible specify how many valid signatures must present for each collection item.

Reviewer: ```suggestion
With ``--galaxy-keyring`` set, it is possible to specify how many valid signatures must be present for each collection item.
```

> Bear in mind that if this is set to an number collections having no signatures at all still pass.

Reviewer: ```suggestion
```

I'd prefer to leave discussion of the details of what ansible-galaxy does with this value to the galaxy docs.

> The value is passed unmodified to ansible-galaxy calls via the option

Reviewer: ```suggestion
The value is passed unmodified to ``ansible-galaxy`` calls via the option
```

> --galaxy-required-valid-signature-count. See the ansible-galaxy doc for more info.

Reviewer: ```suggestion
``--required-valid-signature-count``. See the ``ansible-galaxy`` documentation for more information.
```
